File: frontend/src/types/stats.ts
export interface SplitStats {
  images: number;
  labels: number;
}

export interface DatasetStats {
  dataset: string;
  total_images: number;
  splits: Record<string, SplitStats>;
  num_labels: number;
  num_classes: number;
  avg_images_per_class: number;
  missing_labels: number;
  missing_images: number;
  duplicate_images: number;
  duplicate_labels: number;
  corrupted_files: number;
  empty_label_files: number;
}

export interface ValidationIssue {
  check: string;
  severity: "error" | "warning" | "info";
  split?: string | null;
  file?: string | null;
  line_number?: number | null;
  message: string;
  details: Record<string, unknown>;
}

export interface ValidationReport {
  dataset: string;
  issues: ValidationIssue[];
  summary: Record<string, number>;
}

File: backend/app/schemas/stats.py
from pydantic import BaseModel


class SplitStats(BaseModel):
    images: int
    labels: int


class DatasetStats(BaseModel):
    dataset: str
    total_images: int
    splits: dict[str, SplitStats]
    num_labels: int
    num_classes: int
    avg_images_per_class: float
    missing_labels: int
    missing_images: int
    duplicate_images: int
    duplicate_labels: int
    corrupted_files: int
    empty_label_files: int
